Do not run "download" and "before build" section for resume

junaruga opened the pull request:
This fixes https://github.com/sclorg/rpm-list-builder/issues/11

We should not run those section when it is the resume mode, because the resume mode is expected to run after users finish "download" and "before build" process, fail on "build" process.

Below are the result to run the command.
We can see the "before build" process is not run and skipped as the result does not include "Package was successfully deleted.".

```
$ rpmlb \
  --build custom \
  --custom-file tests/fixtures/custom/rhpkg_internal_copr.yml \
  --work-directory $HOME/tmp/rpmlb_work_dir \
  ../rhscl-rebuild-recipes/ror.yml \
  --resume 9 \
  rh-ror50 2>&1 | tee rpmlb.log
2017-05-03 15:05:26,437 INFO     rpmlb.app                  Starting rpmlb (1.0.0)
2017-05-03 15:05:26,440 INFO     rpmlb.work                 Working directory: /home/jaruga/tmp/rpmlb_work_dir
2017-05-03 15:05:26,441 INFO     rpmlb.app                  Downloading...
2017-05-03 15:05:26,441 INFO     rpmlb.downloader.base      Skip the download process for resume option.
2017-05-03 15:05:26,441 INFO     rpmlb.app                  Building...
2017-05-03 15:05:26,441 INFO     rpmlb.builder.base         Skip the process before build for resume option.
...
Wrote: /home/jaruga/tmp/rpmlb_work_dir/009/rubygem-flexmock/rubygem-flexmock-2.3.4-2.el7.src.rpm
...
```


Can you check this PR?
Thanks.

junaruga commented:
I found a mistake in the test code. Just moment.

junaruga commented:
Amended. Ready for review.

khardix commented on rpmlb/downloader/base.py:
```diff
@@ -38,6 +38,11 @@ def get_instance(cls, name):
         return instance
 
     def run(self, work, **kwargs):
+        is_resume = 'resume' in kwargs and kwargs['resume']
```
There is a better python idiom for that:
```python
is_resume = kwargs.get('resume', False)
```

The `dict.get(key, default)` returns the value with `key` if such key is in the dictionary, else it returns the `default` value. [docs](https://docs.python.org/3/library/stdtypes.html?highlight=dict.get#dict.get)

junaruga replied:
Thanks. It looks better. I will modify with your way.

hroncok commented on rpmlb/builder/base.py:
```diff
@@ -43,13 +43,16 @@ def get_instance(cls, name):
             resume_num = kwargs['resume']
 
+        if is_resume:
+            LOG.info('Skip the process before build for resume option.')
```
It's really hard to understand that sentence.

junaruga replied:
@hroncok is there any suggestion for the sentence?

hroncok replied:
...because the resume option was used?

junaruga replied:
You mean "Skip the process before build, because the resume option was used"?
OK, I will modify it with your way.

hroncok replied:
That's how I understood it. I hope I'm right.

junaruga replied:
Your recognition is right. I think that your sentence is better than my sentence to understand easily and accurately.

junaruga commented:
I fixed mentioned items for the review.
I added issue URL to commit log.
I did split the target tests. and changed the test names to be self-explanatory.

And I changed below implementation to use `is_resume = kwargs.get('resume', False)`.
The situation is

In `rpmlb/app.py` `_normalize_args_to_dict` function

```
args_dict = vars(args)
```

`args_dict` elements included even not specified options as value is `None` by `vars`.
The value was like below.

```
2017-05-04 17:04:22,264 INFO     rpmlb.app                  args_dict: {'recipe_file': '/home/jaruga/git/rpm-build-tool/rpm-list-builder/tmp/recipes/ror_test_macro.yml', 'recipe_id': 'rh-ror50', 'verbose': False, 'download': 'local', 'build': 'dummy', 'work_directory': '/home/jaruga/git/rpm-build-tool/rpm-list-builder/tmp/work_directory', 'branch': None, 'source_directory': '/home/jaruga/git/rpm-build-tool/rpm-list-builder/tmp/source_directory', 'resume': None, 'mock_config': None, 'copr_repo': None, 'custom_file': None}
```
This elements that have value: `None` are problem to use `is_resume = kwargs.get('resume', False)`.
Below is the sample code.

```
$ cat get_from_dict.py 
d = {}
is_resume = d.get('resume', False)
print('1: ' + str(is_resume))

d = {'resume': None}
is_resume = d.get('resume', False)
print('2: ' + str(is_resume))

d = {'resume': True}
is_resume = d.get('resume', False)
print('3: ' + str(is_resume))
```

```
$ python3 get_from_dict.py
1: False
2: None
3: True
```

So, I removed the dirty and meaningless elements to normalize the `args_dict` object in the initial process.

Amended. Ready for the review.

junaruga commented:
Ah just moment.

junaruga commented:
Amended. Ready for the review.

hroncok commented:
Rebase and later merge I guess. Open issues for any remaining concerns.

junaruga commented:
Rebase for this PR? OK I will do it.

junaruga commented:
Rebased. But sorry I need review again for "Review required". Maybe I canceled the review flag by myself.

junaruga commented:
> Open issues for any remaining concerns.

I guess I have written all the concerns to the issues.

hroncok commented:
The merging rules are very strict now. We can talk about it an maybe lower them a bit down.

junaruga commented:
> The merging rules are very strict now. We can talk about it an maybe lower them a bit down.

I see. I prefer a little bit loose option keeping review accepted flag after the rebase.

junaruga commented:
@khardix sorry I need your approval again for this PR.
